pkg/policies/controlplane/components/switcher.go

```diff
@@ -39,7 +40,7 @@ func (dec *Switcher) Execute(inPortReadings runtime.PortToReading, tickInfo runt
 
 	var output runtime.Reading
 
-	if switchValue.Valid() && switchValue.Value() != 0 {
+	if tristate.ReadValue(switchValue) > 0 {
```

Comment:
This looks very incorrect. Earlier we were reading actual value. Tristate "ReadValue" (renamed recently) is not reading value but instead, it's telling whether the reading is valid or not. Is this a correct behavior?
We also need to compare against tristate.Bool -- i.e. understand Unknown/true/false values. This comparison will cause the program to enter the if condition when tristate is Unknown and that was not the case earlier -- earlier we returned from this Execute method when the reading was invalid.

Reply:
shifted to tristate.True